src/mme-app/utils/mmeCauseUtils.cpp

```diff
@@ -0,0 +1,86 @@
+
+/*
+ * Copyright 2020-present Infosys Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <controlBlock.h>
+#include <s1ap_structs.h>
+#include <contextManager/subsDataGroupManager.h>
+#include <log.h>
+#include <utils/mmeCauseUtils.h>
+#include <utils/mmeCauseTypes.h>
+
+using namespace mme;
+
+uint32_t MmeCauseUtils::convertToNasEmmCause(MmeErrorCause mmeErrorCause)
+{
+    log_msg(LOG_DEBUG, "MME Error Cause %d\n", mmeErrorCause);
+
+	uint32_t nasEmmCause = 0;
+
+	switch(mmeErrorCause)
+	{
+		case secModeRespFailure_c:
+		case s6AiaFailure_c:
+			nasEmmCause = emmCause_ue_id_not_derived_by_network;
+			break;
+
+		default:
+			nasEmmCause = emmCause_network_failure;
+			break;
+	}
+	return nasEmmCause;
+}
+
+uint32_t MmeCauseUtils::convertToGtpCause(MmeErrorCause mmeErrorCause)
+{
+    log_msg(LOG_DEBUG, "MME Error Cause %d\n", mmeErrorCause);
+
+	uint32_t gtpCause = 0;
+
+	switch (mmeErrorCause)
+	{
+		case networkTimeout_c:
+			gtpCause = 0x87; //UE not responding
```

Comment:
Can you put these in some correct .h file ?

Reply:
Done.